### DISCUSSION TITLE
The remote certificate was rejected by the provided RemoteCertificateValidationCallback.

### DISCUSSION THREAD

**Discussion by varshadqz** [dsbenghe/Novell.Directory.Ldap.NETStandard#187] [Q&A]
I created the LDAP SSL secure connection on the AD domain server. Using the `ldp.exe` tool (on the same AD server) I am able to connect with the SSL. This means LDAPS is enabled on the AD server. Now I want to connect this AD in Asp.Net Core  5 Application. Trying following code:
```
public LdapConnection GetLDAPConnection(IOptions<ADConfiguration> _settings)
{
    LdapConnection connection = new LdapConnection { SecureSocketLayer = true };
    connection.Connect(_settings.Value.DomainIPAddress, _settings.Value.Port); //port is 636
    connection.Bind(_settings.Value.AdminDn, _settings.Value.Password);

    if (connection.Bound)
    {
        return connection;
    }

    return null;
}
```

It is throwing an exception:

> 
System.Security.Authentication.AuthenticationException: 'The remote certificate was rejected by the provided RemoteCertificateValidationCallback.'

I am not getting if I can connect using ldp.exe then why not with the C# code? do I need any CA certificate at the client-side also?? Please help. I have also posted this question here https://stackoverflow.com/questions/70361557/c-sharp-cannot-connect-to-ad-using-ldaps

**Comment by dsbenghe** [comment]
The error is not the best - it should get fixed in 4.0. But this means that the certificate of the server failed validation.

You can register your own callback and you can see what is the error more clearly.
